File: vulcanus/src/main/java/com/merkury/vulcanus/model/dtos/spot/HomePageSpotDto.java
package com.merkury.vulcanus.model.dtos.spot;

import com.merkury.vulcanus.model.embeddable.BorderPoint;
import lombok.Builder;

import java.util.Set;

@Builder
public record HomePageSpotDto(Long id,
                              String name,
                              Double rating,
                              Integer ratingCount,
                              String firstPhoto,
                              Set<SpotTagDto> tags,
                              BorderPoint centerPoint,
                              String city,
                              Double distanceToUser) {
}

File: venus/src/model/interface/spot/search-spot/homePageSpotDto.ts
import SpotCoordinatesDto from "../coordinates/spotCoordinatesDto";
import TagDto from "../../tagDto";

export default interface HomePageSpotDto {
    id: number;
    name: string;
    rating: number;
    ratingCount: number;
    firstPhoto: string;
    centerPoint: SpotCoordinatesDto;
    tags: TagDto[];
    city: string;
    distanceToUser: number;
}
